packages/core/src/amazonqGumby/telemetry/codeTransformTelemetry.ts

```diff
@@ -90,3 +105,79 @@ export const javapOutputToTelemetryValue = (javapCommandLineOutput: string) => {
 }
 
 export const calculateTotalLatency = (startTime: number): number => Date.now() - startTime
+
+type CodeTransformAPIresponse =
+    | CreateUploadUrlResponse
+    | ResumeTransformationResponse
+    | StopTransformationResponse
+    | GetTransformationPlanResponse
+    | StartTransformationResponse
+    | GetTransformationResponse
+
+export class TelemetryHelper {
+    private static get sessionId() {
+        return CodeTransformTelemetryState.instance.getSessionId()
+    }
+
+    static async callApi<T extends CodeTransformAPIresponse>(parameters: {
+        apiCall: () => Promise<PromiseResult<T, AWSError>>
+        apiName: CodeTransformApiNames
+        errorReason: string
+        uploadId?: string
+        jobId?: string
+        uploadFileByteSize?: number
+        setJobFailureMetadata: boolean
```

Comment:
We can default this to `true`? I'm not sure why GetTransformation was the only operation with setJobFailureMetadata set to false, but I think it can also benefit from the metadata (requestId)